stdlib/n-ary-app.kl

```diff
@@ -109,23 +109,47 @@
                                           (cons-list-syntax 'my-arrow args stx))
                                         stx))])]))]))
 
-(export (rename ([my-arrow ->]) my-arrow))
+(meta
```

Comment:
I don't think this is the right place for this code, as it's not related to nary application. Perhaps we could have a new module, and re-export these identifiers from `prelude.kl`?

Reply:
Good point. Moved to `stdlib/builtins.kl`.

`n-ary-app.kl` kind of became our unofficial prelude, and so it re-exports a ton of stuff which have nothing to do with n-ary functions, and a ton of examples use `#lang "n-ary-app.kl"`. I've changed it to only export its new macros, and I've changed all the examples to use `#lang "prelude.kl"` instead.